sonar-db/src/main/java/org/sonar/db/version/v55/FeedRulesTypes.java

```diff
@@ -0,0 +1,105 @@
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.db.version.v55;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Joiner;
+import com.google.common.base.Splitter;
+import java.sql.SQLException;
+import java.util.List;
+import org.sonar.api.utils.System2;
+import org.sonar.core.rule.RuleType;
+import org.sonar.db.Database;
+import org.sonar.db.version.BaseDataChange;
+import org.sonar.db.version.MassUpdate;
+import org.sonar.db.version.MassUpdate.Handler;
+import org.sonar.db.version.Select.Row;
+import org.sonar.db.version.SqlStatement;
+
+import static com.google.common.collect.Lists.newArrayList;
+import static org.apache.commons.lang.StringUtils.defaultString;
+
+/**
+ * Duplicates RuleTagsToTypeConverter from API on purpose. Db migration must be isolated and must not
+ * depend on external code that may evolve over time.
+ *
+ * https://jira.sonarsource.com/browse/MMF-141
+ */
+public class FeedRulesTypes extends BaseDataChange {
+
+  private static final Splitter TAG_SPLITTER = Splitter.on(',');
+  private static final Joiner TAG_JOINER = Joiner.on(',').skipNulls();
+  private final System2 system;
+
+  public FeedRulesTypes(Database db, System2 system) {
+    super(db);
+    this.system = system;
+  }
+
+  @Override
+  public void execute(Context context) throws SQLException {
+    MassUpdate update = context.prepareMassUpdate().rowPluralName("rules");
+    update.select("SELECT id, system_tags, tags FROM rules WHERE rule_type IS NULL OR rule_type=0");
+    update.update("UPDATE rules SET rule_type=?, system_tags=?, tags=?, updated_at=? WHERE id=?");
+    update.execute(new MigrationHandler(system.now()));
+  }
+
+  private static final class MigrationHandler implements Handler {
+
+    private final long now;
+
+    private MigrationHandler(long now) {
+      this.now = now;
+    }
+
+    @Override
+    public boolean handle(Row row, SqlStatement update) throws SQLException {
+      long id = row.getLong(1);
+
+      // See algorithm to deduce type from tags in RuleTagsToTypeConverter
+      List<String> systemTags = newArrayList(TAG_SPLITTER.split(defaultString(row.getNullableString(2))));
+      List<String> userTags = newArrayList(TAG_SPLITTER.split(defaultString(row.getNullableString(3))));
+      RuleType type = tagsToType(systemTags);
+      systemTags.remove("bug");
+      systemTags.remove("security");
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "security" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1192)